app/controllers/amr_heatmap_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AmrHeatmapController < ApplicationController
+  include ReportHelper
+
+  before_action :admin_required
+
+  current_power do # Put this here for CLI
+    Power.new(current_user)
+  end
+
+  power :viewable_samples, as: :samples_scope
+
+  # first we grab the Sample from the id parameter passed to us
+  before_action :set_sample
+
+  # GET /amr_heatmap.json
+  # Return JSON information required to create a visualization,
+  # from submitted parameters
+  def index
```

Comment:
It's not clear that this function should be the index. I would have expected the `index` to be the controller that renders the view for the amr heatmap. Maybe rename this function to `get_amr_counts`.

Reply:
@MarkAZhang, would it make sense for this to be the `show` endpoint?
we should avoid the `get_` since that is normally implicit on the GET action.

Reply:
I changed the endpoint to `amr_counts`, does that work?

Reply:
SGTM. Don't think `show` makes sense here since that's normally reserved for a view (e.g. `show.html.erb`).